database-plugins/src/main/java/co/cask/hydrator/plugin/ETLDBOutputFormat.java

```diff
@@ -61,15 +62,44 @@
       Connection connection = getConnection(conf);
       PreparedStatement statement = connection.prepareStatement(constructQuery(tableName, fieldNames));
       return new DBRecordWriter(connection, statement) {
+
+        private boolean emptyData = true;
+
         @Override
         public void close(TaskAttemptContext context) throws IOException {
-          super.close(context);
+          try {
+            if (!emptyData) {
+              getStatement().executeBatch();
+              getConnection().commit();
+            }
+          } catch (SQLException e) {
+            try {
+              getConnection().rollback();
+            } catch (SQLException ex) {
+              LOG.warn(StringUtils.stringifyException(ex));
```

Comment:
how come this is done instead of logging the exception with some descriptive message?